R/SoilProfileCollection-setters.R

```diff
@@ -339,14 +339,12 @@ setReplaceMethod("site", signature(object = "SoilProfileCollection"),
   # the horizon data !
   idx <- idx[-match(idname(object), names_attr)]
 
-  # this will break when multiple horizons in the same pedon have different site data!
-  # this seems to work fine in all cases, as we keep the ID column
-  # and it ensures that the result is in the same order as the IDs
   .SD <- NULL
   
   dth <- as.data.table(horizons(object))
+  idn <- dth[[idname(object)]]
   
-  new_site_data <- .as.data.frame.aqp(unique(dth[, .SD, .SDcols = names_attr]), aqp_df_class(object))
+  new_site_data <- .as.data.frame.aqp(dth[, unique(.SD), .SDcols = names_attr, by = list(idn)], aqp_df_class(object))
```

Comment:
Now third-or-fourth guessing this -- I think the current code is safe -- since the profile ID (`idname(object)`) is part of `names_attr`... but this is how you would apply unique to each profile in the horizon data to make site.

Reply:
So, I guess this is just syntax comparison -- apply `unique()` to whole data.table (where idname is a column) versus apply unique on profile-horizon subdataframes with `j =unique(.SD)`